pumpkin-macros/src/lib.rs

```diff
@@ -352,3 +352,62 @@ pub fn block_property(input: TokenStream, item: TokenStream) -> TokenStream {
 
     code.into()
 }
+
+#[proc_macro_derive(PersistentDataHolder, attributes(persistent_data))]
+pub fn derive_persistent(input: TokenStream) -> TokenStream {
+    let input = parse_macro_input!(input as DeriveInput);
+    let name = input.ident;
+
+    let field = if let syn::Data::Struct(data) = &input.data {
+        data.fields.iter().find_map(|f| {
+            for attr in &f.attrs {
+                let path = attr.path();
+                if path.is_ident("persistent_data") {
+                    return f.ident.clone();
+                }
+            }
+            None
+        })
+    } else {
+        None
+    };
+
+    let field = match field {
+        Some(f) => f,
```

Comment:
there's a somewhat new feature you can use that's a little shorter than this match statement

https://doc.rust-lang.org/rust-by-example/flow_control/let_else.html